src/fragmentarium/domain/text.ts

```diff
@@ -140,11 +140,31 @@ export type Token =
   | Enclosure
   | Tabulation
 
-export interface Line {
+export type Line = TextLine | RulingDollarLine
+
+export interface LineBase {
   readonly type: string
   readonly prefix: string
   readonly content: ReadonlyArray<Token>
 }
+export interface LineNumber {
+  readonly number: number
+  has_prime: false
+  prefix_modifier?: string
+  suffix_modifier?: string
+}
+export interface TextLine extends LineBase {
+  readonly line_number?: LineNumber
+}
+
+export interface ControlLines extends LineBase {
+  readonly display_value: string
+}
```

Comment:
You should concentrate on getting https://trello.com/c/CHRE58CU done and not add all the new stuff here. There is no point in splitting the card if you are going to put everything in one huge pull request again. We should use [isolated feature branches](https://www.martinfowler.com/bliki/FeatureBranch.html) and not stick all work in progress in a single branch. It is also faster and easier to review concise pull requests.

Also, the property names are wrong. JavaScript convention is to use camel-case property names. Please consult the schemas for correct names (there are some cases where snake-case is erroneously used). All new code should use proper naming conventions.